packages/protocol/contracts/bridge/Bridge.sol

```diff
@@ -138,42 +180,58 @@ contract Bridge is EssentialContract, IBridge {
         sameChain(message.srcChainId)
     {
         bytes32 msgHash = hashMessage(message);
-        if (isMessageRecalled[msgHash]) revert B_RECALLED_ALREADY();
+        if (messageStatus[msgHash] != Status.NEW) revert B_RECALLED_NOT_ALLOWED();
 
-        ISignalService signalService = ISignalService(resolve("signal_service", false));
-
-        if (!signalService.isSignalSent(address(this), msgHash)) revert B_MESSAGE_NOT_SENT();
-
-        bool received = _proveSignalReceived(
-            signalService, _signalForFailedMessage(msgHash), message.destChainId, proof
-        );
-        if (!received) revert B_NOT_FAILED();
+        bool isMessageNew = messageReception[msgHash].receivedAt == 0;
```

Comment:
`isMessageNew` a bit of a confusing name because it can both mean a message with the status NEW (checked just before this), or a message that already has its message hash committed. Maybe `isMessageReceived` or `isMessageCommitted` (but this would require a rename of `messageReception` to `messageCommitments` to make that work well I think).

Reply:
I kind of like `messageProofs`/`isMessageProven` more than `messageCommitments` that I first thought would be a good name. If nobody else comes up with a better name/has any objections in the next couple of hours I'll do the rename in this PR!